src/Services/IReferenceTokenValidator.cs

```diff
@@ -0,0 +1,11 @@
+using Duende.IdentityServer.EntityFramework.DbContexts;
+using IdentityServerConfig.Models;
+
+namespace IdentityServerConfig.Services;
+
+public interface IReferenceTokenValidator
+{
+    bool IsActive(DateTime? expirationDate);
```

Comment:
I think we should either remove this `IsActive` method and move back the implementation to the View, or we need to accept a `CheckReferenceTokenModel` as a parameter here. Right now the `IsActive` method has no specific relation ship with a reference token as we're only passing in an expiration date which could be applicable to pretty much anything.